tests/jballerina-unit-test/src/test/resources/test-src/types/intersection/test_intersection_type.bal

```diff
@@ -263,6 +263,28 @@ function testRuntimeTypeNameOfIntersectionType() {
 
 type Error error<record { string message; }>;
 
+type READONLY_REF1 readonly;
+type READONLY_REF2 readonly;
+type READONLY_REF3 readonly;
+
+type ReadonlyObjectType1 object {
+    int atb1;
+    function call() returns int;
+};
+
+type IntersectionType1 READONLY_REF3 & record {|
+    int a;
+|};
+
+type FooType record {|
+    int a;
+|};
+
+type IntersectionType2 READONLY_REF3 & FooType;
+type IntersectionType3 FooType & READONLY_REF3;
+type IntersectionType4 READONLY_REF3 & ReadonlyObjectType1;
+type IntersectionType5 ReadonlyObjectType1 & READONLY_REF3;
+
```

Comment:
Can we add at least one assertion?

Reply:
updated in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/42025/commits/2078b44f32ee4761c82de5c280a9321132a18dda).